src/game-event.ts

```diff
@@ -194,13 +195,31 @@ export interface GameEvents {
     cx: string; // hat999 // TODO: cosmetics key?
   };
   api_response:
-    | any
+    // https://www.typescriptlang.org/docs/handbook/2/narrowing.html#discriminated-unions
+    // | { type: string }
+    // TODO: there is a lot of different types of api responses, depending on api call
```

Comment:
Maybe this can be taken from the `ServerToClient` socket events in some way? I'm not sure. If it's not the case, I'd suggest we move this type into its own file so we can have one type per possible response and then `BetterUX` them.

Reply:
Hmm I think there is a difference from socket events and the response from an api call, also unsure if it gets modified on the way to the response, will have to investigate

Reply:
The importing thing here is the type descrimination so Typescript resolves the correct response type.

Reply:
@Telokis If we used BetterUX would we still get this behaviour dependant on type though? As seen here it has used discriminated unions to know the correct type based on the if statement.
![image](https://user-images.githubusercontent.com/9084377/209372266-118e0410-40a5-4360-b0b4-10ca7abe223b.png)

Reply:
Yes of course. And it will make it even more comfortable to use: you'll be able to rely on any property instead of only `type` as long as they are unique the narrowing would be as expected.